### DISCUSSION TITLE
No Pynq driver generated

### DISCUSSION THREAD

**Discussion by dprzewlocka** [Xilinx/finn#530] [Q&A] [answered]
Hello! I have a strange problem generating the Pynq driver:

I use the variation :) of the "Building the Streaming Dataflow Accelerator" notebook but I only generate the streamlined onnx model and then I do folding with my own parameters in an 'old way' (for some reason the folding parameters that are automatically generated violate the FINN rules)

- To generate the streamlined model I use the following configuration:

```
cfg_estimates = build.DataflowBuildConfig(
  output_dir  = estimates_output_dir,
  mvau_wwidth_max  = 80,
  target_fps = 100,
  synth_clk_period_ns = 10.0,
  board ="ZCU104",
  steps  = build_cfg.estimate_only_dataflow_steps,
  generate_outputs=[
      build_cfg.DataflowOutputType.ESTIMATE_REPORTS,
      build_cfg.DataflowOutputType.PYNQ_DRIVER,
  ]
)
```

- Then I take the step_target_fps_parallelization.onnx model and do folding, and finally I generate the hardware

- But somehow my synth onnx model does not contain the path to Pynq driver, the metadata_props are as follows:

```
[key: "floorplan_json"
value: "/tmp/finn_dev_vision/vitis_floorplan_e1rc2ta2/floorplan.json"
, key: "vivado_pynq_proj"
value: "/tmp/finn_dev_vision/vivado_zynq_proj_m5xd8106"
, key: "bitfile"
value: "/tmp/finn_dev_vision/vivado_zynq_proj_m5xd8106/resizer.bit"
, key: "hw_handoff"
value: "/tmp/finn_dev_vision/vivado_zynq_proj_m5xd8106/resizer.hwh"
, key: "vivado_synth_rpt"
value: "/tmp/finn_dev_vision/vivado_zynq_proj_m5xd8106/synth_report.xml"
, key: "platform"
value: "zynq-iodma"
]
```
What am I missing that the Pynq driver is not generated? Thanks for your help! I'm using the FINN v0.7.

**Comment by rpitonak** [comment, marked as answer]
Hello,

Maybe this answer can help solve your problem: https://github.com/Xilinx/finn/discussions/442#discussioncomment-1675720
